src/stanza.c

```diff
@@ -431,7 +431,7 @@ int xmpp_stanza_to_text(xmpp_stanza_t *stanza,
     ret = _render_stanza_recursive(stanza, buffer, length);
     if (ret < 0) return ret;
 
-    if (ret > length - 1) {
+    if ((unsigned)ret > length - 1) {
```

Comment:
What I meant is that `size_t` can be `unsigned long` instead of `unsigned int` or even something else. It will have different size on different architectures/OSes. For example, in linux 64bit:
```
#include <stddef.h> /* size_t */
#include <stdio.h>

int main()
{
	printf("size_t == %zu bytes\n", sizeof(size_t));
	printf("unsigned == %zu bytes\n", sizeof(unsigned));

	return 0;
}

$ ./a.out
size_t == 8 bytes
unsigned == 4 bytes
```
I believe in windows 64bit it will be 4 bytes, etc.
P.S. I saw such a code somewhere:
```
typedef size_t ssize_t
```